src/parser/xasl_generation.c

```diff
@@ -2931,6 +2997,109 @@ pt_split_hash_attrs (PARSER_CONTEXT * parser, TABLE_INFO * table_info, PT_NODE *
   return ER_FAILED;
 }
 
+/*
+ * pt_split_hash_attrs_for_HQ () - Split the attr_list into two lists without destroying
+ *      the original list for HQ
+ *   return:
+ *   parser(in):
+ *   pred(in):
+ *   build_attrs(out):
+ *   probe_attrs(out):
+ *
+ * Note :
+ * is_PRIOR | NAME_without prior | characteristic
+ *    O     |        O           | unhashable
+ *    O     |        X           | probe attr
+ *    X     |        O           | build attr
+ *    X     |        X           | constant (can be probe or build attr)
+ */
+static int
+pt_split_hash_attrs_for_HQ (PARSER_CONTEXT * parser, PT_NODE * pred, PT_NODE ** build_attrs, PT_NODE ** probe_attrs)
+{
+  PT_NODE *node = NULL, *save_node = NULL, *save_next = NULL;
+  PT_NODE *arg1 = NULL, *arg2 = NULL;
+
+  assert (build_attrs != NULL && *build_attrs == NULL);
+  assert (probe_attrs != NULL && *probe_attrs == NULL);
+  *build_attrs = NULL;
+  *probe_attrs = NULL;
+
```

Comment:
*build_attrs = NULL;
 *probe_attrs = NULL;
They are already checked as "not null" by assert so the codes looks redundant.

Reply:
I think it would be fine to keep it for releases. actually, it is the code generated through copy and paste. since this function is invoked from only one, assert and init may not be necessary. But it looks good to keep it for later use.